skrub/_agg_joiner.py

```diff
@@ -145,225 +126,176 @@ class AggJoiner(BaseEstimator, TransformerMixin):
+    def _check_dataframes(self, X, aux_table):
+        """Check dataframes input types.
+
+        Raises an error if frames aren't both Pandas or Polars dataframes,
+        or if there is a Polars lazyframe. Alternatively, allows `aux_table` to be "X".
 
         Parameters
         ----------
         X : DataframeLike
-            Input data, based table on which to left join the
-            auxiliary tables.
-
-        y : array-like of shape (n_samples), default=None
-            Prediction target. Used to compute correlations between the
-            generated covariates and the target for screening purposes.
-
-        Returns
-        -------
-        AggJoiner
-            Fitted :class:`AggJoiner` instance (self).
+            The main table to augment.
+        aux_table : DataframeLike or "X"
+            The auxiliary table.
         """
-        self.check_input(X)
-        skrub_px, _ = get_df_namespace(*self.aux_table_)
-
-        num_operations, categ_operations = split_num_categ_operations(self.operation_)
+        # Polars lazyframes will raise an error here.
+        if not hasattr(X, "__dataframe__"):
+            raise TypeError(f"'X' must be a dataframe, got {type(X)}.")
+        if isinstance(aux_table, str):
+            if aux_table == "X":
+                return X, X
+            else:
+                raise ValueError("'aux_table' must be a dataframe or 'X'.")
+        if not hasattr(aux_table, "__dataframe__"):
+            raise TypeError(
+                f"'aux_table' must be a dataframe or 'X', got {type(aux_table)}."
+            )
 
-        aux_tables = []
-        for aux_table, aux_key, cols, suffix in zip(
-            self.aux_table_, self.aux_key_, self.cols_, self.suffix_
+        if (is_pandas(X) and not is_pandas(aux_table)) or (
+            is_polars(X) and not is_polars(aux_table)
         ):
-            aux_table = skrub_px.aggregate(
-                aux_table,
-                aux_key,
-                cols,
-                num_operations,
-                categ_operations,
-                suffix=suffix,
+            raise TypeError(
+                "'X' and 'aux_table' must be of the same type, got"
+                f"{type(X)} and {type(aux_table)}"
             )
-            aux_table = self._screen(aux_table, y)
-            aux_tables.append((aux_table, aux_key))
-        self.aux_table_ = aux_tables
 
-        return self
-
-    def transform(self, X):
-        """Left-join pre-aggregated tables on `X`.
+        return X, aux_table
 
-        Parameters
-        ----------
-        X : DataFrameLike
-            The input data to transform.
+    def _check_operations(self):
+        """Check `operations` input.
 
         Returns
         -------
-        X_transformed : DataFrameLike
-            The augmented input.
+        operations
+            1-dimensional array of operations to perform on columns.
         """
+        operations = atleast_1d_or_none(self.operations)
```

Comment:
Yes it was inherited from what existed previously, I'll remove it in the `AggJoiner`. It can be removed later as it's just used in the `AggTarget` and in some utils that will be rewritten or replaced by dispatch